leader-election-impl/src/main/java/com/palantir/paxos/PaxosStateLogImpl.java

```diff
@@ -235,10 +236,24 @@ public void truncate(long toDeleteInclusive) {
         }
     }
 
+    @Override
+    public void truncateAllRounds() {
+        lock.writeLock().lock();
+        try {
+            File dir = new File(path);
+            boolean success = getLogEntries(dir).stream().map(File::delete).reduce(true, (x, y) -> x && y);
+            if (!success) {
+                log.warn("failed to delete all log files.");
+            }
+        } finally {
+            lock.writeLock().unlock();
+        }
+    }
+
     private List<File> getLogEntries(File dir) {
         File[] files = dir.listFiles();
         if (files == null) {
-            return null;
+            return ImmutableList.of();
```

Comment:
I realise this case is handled already above in `getExtremeLogEntry`, but we should clean it up.